### PR TITLE
Bump Typescript version to 5.3.3 

### PR DESCRIPTION
**Pull request by marmoure**
Updates Typescript version and eslint, also fix linting error and code issues.

### CODE REVIEW

File: prosemirror-lwdita/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (adamretter) [inline prosemirror-lwdita/.eslintrc.json:1]: I think for the most part this should be the same as https://github.com/evolvedbinary/lwdita/blob/main/.eslintrc
Although you don't need to add the `eslint-plugin-notice` in this PR

File: prosemirror-lwdita/package.json [hunk 2]
```diff
@@ -66,19 +66,18 @@
     "@tsconfig/recommended": "^1.0.6",
     "@types/chai": "^4.3.11",
     "@types/mocha": "^10.0.6",
-    "@typescript-eslint/eslint-plugin": "^2.30.0",
-    "@typescript-eslint/parser": "^2.30.0",
+    "@typescript-eslint/eslint-plugin": "^7.4.0",
```
Comment (adamretter) [inline prosemirror-lwdita/package.json:69]: Should be `^7.11.0` (same as lwdita project)

File: prosemirror-lwdita/package.json [hunk 3]
```diff
@@ -66,19 +66,18 @@
...
+    "eslint": "^8.57.0",
     "mocha": "^10.2.0",
     "nodemon": "^3.0.2",
     "nyc": "^15.1.0",
     "rimraf": "^5.0.5",
     "ts-node": "^9.1.1",
     "typedoc": "^0.25.7",
     "typedoc-plugin-missing-exports": "^2.2.0",
-    "typescript": "4.6.3"
+    "typescript": "5.3.3"
```
Comment (adamretter) [inline prosemirror-lwdita/package.json:81]: Should be `5.4.5` (same as lwdita project)

File: prosemirror-lwdita/src/commands.ts [hunk 4]
```diff
@@ -20,8 +19,6 @@ export function createNode(type: NodeType, args: Record<string, any> = {}): Node
   switch (type.name) {
     case 'p': return type.createAndFill() as Node;
     case 'data': return type.createAndFill({}, type.schema.text('text')) as Node;
-    case 'stentry': return type.createAndFill({}, createNode(type.schema.nodes['p'])) as Node;
```
Comment (adamretter) [inline prosemirror-lwdita/src/commands.ts:23]: Why have these been removed?

File: prosemirror-lwdita/src/commands.ts [hunk 5]
```diff
@@ -561,16 +557,15 @@ export function getTree(pos: ResolvedPos, depth = 0) {
  *
  * @param state - The EditorState object
  * @param dispatch - A function to be used to dispatch a transaction
- * @param view - The EditorView object
  * @returns Boolean
  */
-export function enterPressed(state: EditorState, dispatch?: (tr: Transaction) => void, view?: EditorView) {
-  let { $from, empty } = state.selection;
+export function enterPressed(state: EditorState, dispatch?: (tr: Transaction) => void) {
+  let { $from } = state.selection;
```
Comment (adamretter) [inline prosemirror-lwdita/src/commands.ts:563]: Why does one of these have a `$` prefix?

File: prosemirror-lwdita/src/document.ts [hunk 6]
```diff
@@ -112,10 +112,9 @@ export const NODES: Record<string, (value: JDita, parent: JDita) => any> = {
  * Transforms the JDita document into a proper ProseMirror document
  *
  * @param value - The JDita node
- * @param parent - The parent JDita node
  * @returns The transformed JDita node
  */
-function defaultTravel(value: JDita, parent: JDita): any {
+export function defaultTravel(value: JDita): any {
```
Comment (adamretter) [inline prosemirror-lwdita/src/document.ts:117]: Why does this now need to be `exported`?

**Review by adamretter — CHANGES_REQUESTED** [review]

File: prosemirror-lwdita/src/commands.ts (hunk 5 above)
Comment (marmoure) [reply prosemirror-lwdita/src/commands.ts:563]: corresponding to https://prosemirror.net/docs/ref/#state.Selection.$from.

File: prosemirror-lwdita/src/document.ts (hunk 6 above)
Comment (marmoure) [reply prosemirror-lwdita/src/document.ts:117]: It does not, this was an old change before we introduced the escape hatch 
```ts
//Escape hatch for Unit Testing due to a lack of “package-private” accessibility scope in TypeScript
export const _test_private_document = {
  deleteUndefined,
  defaultTravel,
  travel,
}
```
Good catch @adamretter

File: prosemirror-lwdita/src/commands.ts (hunk 4 above)
Comment (marmoure) [reply prosemirror-lwdita/src/commands.ts:23]: it's duplicated we had 2 entries for  `stentry` and `strow`

**Pull request merged**